src/main/java/com/epam/grid/engine/service/HealthCheckProviderService.java

```diff
@@ -55,27 +51,15 @@ public HealthCheckProviderService(@Value("${grid.engine.type}") final EngineType
     /**
      * This method passes the request on to {@link HealthCheckProvider}
      * and returns working grid engine status information.
+     *
      * @return {@link HealthCheckInfo}
      */
     public HealthCheckInfo checkHealth() {
         return getProvider().checkHealth();
     }
 
-    /**
-     * This method injects all created {@link HealthCheckProvider} beans.
-     * @param providers list of existing HealthCheckProviders
-     * @see HealthCheckProvider
-     */
-    @Autowired
-    public void setProviders(final List<HealthCheckProvider> providers) {
-        this.providers = providers.stream()
-                .collect(Collectors.toMap(HealthCheckProvider::getProviderType, Function.identity()));
-    }
-
     private HealthCheckProvider getProvider() {
```

Comment:
There is no need in this method anymore: in case of missing candidate for wiring as `healthCheckProvider`, application will fail to start:

> Field healthCheckProvider in com.epam.grid.engine.service.HealthCheckProviderService required a bean of type 'com.epam.grid.engine.provider.healthcheck.HealthCheckProvider' that could not be found.

So you can omit `getProvider()` and use `healthCheckProvider` in `checkHealth()` directly.
The same logic could be applied to other provider services

Reply:
classes in service package were changed